src/components/discovery.js

```diff
@@ -1,94 +1,91 @@
-        checkInterval: 4000,
-        nodeTimeout: 5000,
-        masterTimeout: 6000,
-        monitor: false,
-        log: true,
-        helloLogsEnabled: true,
-        statusLogsEnabled: true,
-        ignoreProcess: false
-    });
-
-    _.defaults(advertisement, {
-        type: 'service'
-    });
-
-    var d = new Discover(options);
-
-    d.me.processId = d.broadcast.processUuid;
-    d.me.processCommand = process.argv.slice(1).map(function(n) {
-        return n.split('/').slice(-2).join('/');
-    }).join(' ');
-
-    d.advertise(advertisement);
-
-    options.log && log(helloLogger(d));
-
-    d.on('added', function(obj) {
-        if (!options.monitor && obj.advertisement.key != advertisement.key) return;
-
-        options.log && options.helloLogsEnabled && log(statusLogger(obj, 'online'));
-    });
-
-    d.on('removed', function(obj) {
-        if (!options.monitor && obj.advertisement.key != advertisement.key) return;
-
-        options.log && options.statusLogsEnabled && log(statusLogger(obj, 'offline'));
-    });
-
-    return d;
-};
-
-Discovery.setDefaults = function(options) {
-    Discovery.defaults = options;
-};
-
-var helloLogger = function(d) {
-    var adv = d.me,
-        log = [];
-
-    d.me.id = d.broadcast.instanceUuid;
-
-    log.push('\nHello! I\'m'.white);
-    log = log.concat(statusLogger(adv));
-    log.push('\n========================\n'.white);
-
-    return log;
-};
-
-var statusLogger = function(obj, config) {
-    var adv = obj.advertisement,
-        log = [],
-        status;
-
-    switch (config) {
-        case 'online':
-            status = '.online'.green;
-            break;
-        case 'offline':
-            status = '.offline'.red;
-            break;
+const Discover = require('node-discover');
+const colors = require('colors');
+const _ = require('lodash');
+
+class Discovery extends Discover {
+
+    constructor(advertisement, options = {}) {
+        _.defaults(options, Discovery.defaults, {
+            helloInterval: 2000,
+            checkInterval: 4000,
+            nodeTimeout: 5000,
+            masterTimeout: 6000,
+            monitor: false,
+            log: true,
+            helloLogsEnabled: true,
+            statusLogsEnabled: true,
+            ignoreProcess: false,
+        });
+
+        super(options);
+
+        this.advertisement = _.defaults(advertisement, {
+            type: 'service',
+        });
+
+        this.advertise(advertisement);
+
+        this.me.processId = this.broadcast.processUuid;
+        this.me.processCommand = process.argv.slice(1).map((n) => {
+            return n.split('/').slice(-2).join('/');
+        }).join(' ');
+
+        options.log && this.log(this.helloLogger());
+
+        this.on('added', function(obj) {
+            if (!options.monitor &&
+                !_.isEqual(obj.advertisement.key, advertisement.key)) return;
+            options.log && options.helloLogsEnabled &&
+            obj.log(obj.statusLogger('online'));
```

Comment:
Are you sure an `obj` has a `log` method? I believe this line should read `this.log(this.statusLogger('online'));`.